### PR TITLE
pkp/pkp-lib#4953 Ensure an XML parser object is available for data installation

### PR DESCRIPTION
**Pull request by asmecher**

### CODE REVIEW

File: classes/plugins/Plugin.inc.php [hunk 1]
```diff
@@ -512,7 +512,9 @@ function installData($hookName, $args) {
 
 		// Install all data files.
 		foreach($dataFiles as $dataFile) {
-			$sql = $installer->dataXMLParser->parseData($dataFile);
+			$dataXMLParser = new DBDataXMLParser();
```
Comment (defstat) [inline classes/plugins/Plugin.inc.php]: I would have proposed this line and the one after that [515-516] to go above the `foreach`, I don't think that it is needed to be instantiated for each `dataFile`

Comment (asmecher) [reply classes/plugins/Plugin.inc.php]: Yes, thanks! But let me review your other comment before I do that.

**Comment by asmecher** [comment]
> ...in order to follow the same norm as...

Yes, thanks, that's a better solution. I don't like intervening this way with member objects but we're already doing it, so better to stay consistent. I've force-pushed up the change to use what you've suggested instead -- can you take a quick look at it?

File: classes/plugins/Plugin.inc.php (hunk 1 above)
Comment (asmecher) [reply classes/plugins/Plugin.inc.php]: The other comment obsoleted this one.

**Comment by defstat** [comment]
I think we could merge it. Thanks!

**Pull request merged**